app.py

```diff
@@ -53,14 +53,21 @@ def upload_image():
         # Add image to document storage
         document_storage.add_image(file_path)
         
+        
+        global SHOULD_GET_DATA 
+        SHOULD_GET_DATA = False
+        global data 
+        data = ""
+        
         return "File uploaded successfully", 200
 
-@app.route('/analyze', methods=['GET'])
-def analyze_document():
-    document = document_storage.get_document()
-    if not document:
-        return "No documents to analyze", 400
+
+
+def get_data(document):
     
+    global SHOULD_GET_DATA
```

Comment:
Should be defined in the global scope.

Reply:
even if in global scope, must specify the global in function to access the global var